File: frontend_App/todo-list/types/ViewModel/CalendarEventViewModel.ts
interface ICalendarExtendedProps {
  ScheduleId: number
  Summary: string // 概要説明
  UsageTime: number // 使用時間
  CategoryMasterFlg: boolean// カテゴリーマスタを表示するかどうか
  CategoryMasterId: number
}

interface ICalendarEventViewModel {
  Id: number
  AllDay: boolean
  Title: string
  Start: Date
  StartHour: string
  StartMinute: string
  End: Date
  EndHour: string
  EndMinute: string
  TextColor: string
  BackgroundColor: string
  BorderColor: string
  ExtendedProps: ICalendarExtendedProps
}

export class CalendarExtendedProps implements ICalendarExtendedProps {
  ScheduleId = 0
  Summary = ''
  UsageTime = 0
  CategoryMasterFlg = false
  CategoryMasterId = 0
}

export class CalendarEventViewModel implements ICalendarEventViewModel {
  Id = 0
  AllDay = false
  Title = ''
  Start = new Date()
  StartHour = ''
  StartMinute = ''
  End = new Date()
  EndHour = ''
  EndMinute = ''
  TextColor = ''
  BackgroundColor = ''
  BorderColor = ''
  ExtendedProps = new CalendarExtendedProps()
}

File: ToDoApp_backend/ViewModel/Calendar/CalendarEventViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ToDoApp_backend.ViewModel.Calendar
{
    public class CalendarExtendedProps
    {
        public long ScheduleId { get; set; }
        public string Summary { get; set; }
        public int UsageTime { get; set; }
        public bool CategoryMasterFlg { get; set; }
        public long CategoryMasterId { get; set; }
    }

    public class CalendarEventViewModel
    {
        public long Id { get; set; }
        public bool AllDay { get; set; }
        public string Title { get; set; }
        public DateTime? Start { get; set; }
        public string StartHour { get; set; }
        public string StartMinute { get; set; }
        public DateTime? End { get; set; }
        public string EndHour { get; set; }
        public string EndMinute { get; set; }
        public string TextColor { get; set; }
        public string BackgroundColor { get; set; }
        public string BorderColor { get; set; }
        public CalendarExtendedProps ExtendedProps { get; set; }
    }
}
